components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -316,6 +317,60 @@ public IdentityProvider getResidentIdP(String tenantDomain)
         passiveSTSFedAuthn
                 .setProperties(propertiesList.toArray(new Property[propertiesList.size()]));
         fedAuthnCofigs.add(passiveSTSFedAuthn);
+
+        FederatedAuthenticatorConfig sessionTimeoutConfig = IdentityApplicationManagementUtil
+                .getFederatedAuthenticator(identityProvider.getFederatedAuthenticatorConfigs(),
+                        IdentityApplicationConstants.Authenticator.IDPProperties.NAME);
+        if(sessionTimeoutConfig == null){
+            sessionTimeoutConfig = new FederatedAuthenticatorConfig();
+            sessionTimeoutConfig.setName(IdentityApplicationConstants.Authenticator.IDPProperties.NAME);
+        }
+        propertiesList = new ArrayList<Property>(Arrays.asList(sessionTimeoutConfig.getProperties()));
+        if(IdentityApplicationManagementUtil.getProperty(sessionTimeoutConfig.getProperties(),
+                IdentityApplicationConstants.Authenticator.IDPProperties.SESSION_IDLE_TIME_OUT) == null){
+            Property sessionIdletimeOutProp = new Property();
+            sessionIdletimeOutProp.setName(IdentityApplicationConstants.Authenticator.IDPProperties.SESSION_IDLE_TIME_OUT);
+            String idleTimeout = IdentityUtil.getProperty("TimeConfig.SessionIdleTimeout");
+            if(StringUtils.isEmpty(idleTimeout)){
+                idleTimeout = IdentityApplicationConstants.Authenticator.IDPProperties.SESSION_IDLE_TIME_OUT_DEFAULT;
+            }else if(!StringUtils.isNumeric(idleTimeout)){
+                log.warn("SessionIdleTimeout in identity.xml should be a numeric value");
+                idleTimeout = IdentityApplicationConstants.Authenticator.IDPProperties.SESSION_IDLE_TIME_OUT_DEFAULT;
+            }
+            sessionIdletimeOutProp.setValue(idleTimeout);
+            propertiesList.add(sessionIdletimeOutProp);
+        }
+        if(IdentityApplicationManagementUtil.getProperty(sessionTimeoutConfig.getProperties(),
+                IdentityApplicationConstants.Authenticator.IDPProperties.REMEMBER_ME_TIME_OUT) == null){
+            Property rememberMeTimeOutProp = new Property();
+            rememberMeTimeOutProp.setName(IdentityApplicationConstants.Authenticator.IDPProperties.REMEMBER_ME_TIME_OUT);
+            String rememberMeTimeout = IdentityUtil.getProperty("TimeConfig.RememberMeTimeout");
```

Comment:
Introduce an appropriate constant upon the property name "TimeConfig.RememberMeTimeout"